File: alura/php/solid/ContaEstudante.php
<?php
  class ContaEstudante {
    private $manipulador;
    private $milhas;

    public function __construct() {
      $hits->manipulador = new ManipuladorDeSaldo();
    }

    public function deposita($valor) {
      $this->manipulador->deposita($valor);
      $this->milhas += $valor;
    }

    public function getMilhs() {
      return $this->milhas;
    }
  }
?>

File: alura/c-sharp/solid/ContaEstudante.cs
namespace Solid
{
  public class ContaDeEstudante : ContaComum
  {
    private ManipuladorDeSaldo m;
    public int Milhas { get;private set; }

    public ContaDeEstudante()
    {
      m = new ManipuladorDeSaldo()
    }

    public void Deposita(double valor)
    {
      m.Deposita(valor);
      this.Milhas += (int)valor;
    }
  }
}
